components/proxy/src/main/java/com/hotels/styx/startup/ProxyServerSetUp.java

```diff
@@ -0,0 +1,85 @@
+/**
```

Comment:
We need to add some unit tests:

- Notifies plugins when starts.
- Notifies plugins when stops.
- Logs plugin startup failures.
- Attempts to start all plugins before fails.
- Logs plugin shutdown failures.